multiuser/keycloak/che-multiuser-keycloak-server/src/main/java/org/eclipse/che/multiuser/keycloak/server/KeycloakProfileRetriever.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.multiuser.keycloak.server;
+
+import java.io.IOException;
+import java.util.Map;
+import javax.inject.Inject;
+import org.eclipse.che.api.core.ApiException;
+import org.eclipse.che.api.core.ServerException;
+import org.eclipse.che.api.core.rest.HttpJsonRequestFactory;
+import org.eclipse.che.multiuser.keycloak.shared.KeycloakConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Fetches user profile from Keycloack server.
+ *
+ * @author Max Shaposhnik (mshaposh@redhat.com)
+ * @author Sergii Leshchenko
+ */
+public class KeycloakProfileRetriever {
+  private static final Logger LOG = LoggerFactory.getLogger(KeycloakProfileRetriever.class);
+
+  private final String keyclockCurrentUserInfoUrl;
+  private final HttpJsonRequestFactory requestFactory;
+
+  @Inject
+  public KeycloakProfileRetriever(
+      KeycloakSettings keycloakSettings, HttpJsonRequestFactory requestFactory) {
+    this.requestFactory = requestFactory;
+    this.keyclockCurrentUserInfoUrl =
+        keycloakSettings.get().get(KeycloakConstants.USERINFO_ENDPOINT_SETTING);
+  }
+
+  public Map<String, String> retrieveKeycloakAttributes() throws ServerException {
+    Map<String, String> keycloakUserAttributes;
+    // Retrieving own profile
+    try {
+      keycloakUserAttributes =
+          requestFactory.fromUrl(keyclockCurrentUserInfoUrl).request().asProperties();
```

Comment:
You can write `return` here instead of saving it to a local variable and return outside of try block.

Reply:
Sure !
This indeed initially came out of an `Extract Method` refactoring.